File: CodeWars/6 kyu/split-strings/solution.cs
using System;                           // For Console.WriteLine
using System.Linq;                      // For Linq
using System.Collections.Generic;       // For Lists

public class SplitString {
  public static string[] Solution(string str) {
    return Enumerable.Range(0, str.Length)
      .Where(i => i % 2 == 0) // Process even pairs
      .Select(i => $"{str[i]}{(i+1 >= str.Length ? "_" : str[i+1])}")
      .ToArray();
  }
}

File: CodeWars/6 kyu/split-strings/solution.rs
fn solution(s: &str) -> Vec<String> {
    (0..s.len()) // Range from 0 -> length
        .step_by(2) // Process in pairs
        .map(|i| format!("{}{}", &s[i..=i], s.get(i + 1..=i + 1).unwrap_or("_"))) // Format to s[i] + s[i+1] ?? _
        .collect() // Collect into Vec<String>
}
